commons/com.b2international.commons/src/com/b2international/commons/config/ConfigurationFactory.java

```diff
@@ -184,14 +189,46 @@ private void resolveModuleConfigs(Configuration config) throws IOException {
 			final JsonNode moduleConfigNode = config.getModuleConfigurations().get(fieldName);
 			Object moduleConfig = null;
 			if (moduleConfigNode != null) {
+				replaceEnvironmentVariables(moduleConfigNode);
 				moduleConfig = factory.build(moduleConfigNode, fieldName);
 			} else {
 				moduleConfig = factory.build();
 			}
 			config.getConfigurationClassToNode().put(fieldType, moduleConfig);
 		}
 	}
-
+	
+	/**
+	 * Recursively replaces environment variables from a given node.
+	 * 
+	 * @param node
+	 */
+	private void replaceEnvironmentVariables(JsonNode node) {
+		final Iterator<Entry<String, JsonNode>> fields = node.fields();
+		while (fields.hasNext()) {
+			final Entry<String, JsonNode> field = fields.next();
+			
+			final String fieldName = field.getKey();
+			final String textValue = field.getValue().textValue();
+			if (!Strings.isNullOrEmpty(textValue)) {
+				// match ${ENV_VAR_NAME}
+				final Pattern pattern = Pattern.compile("\\$\\{(\\w+)\\}");
+				final Matcher matcher = pattern.matcher(textValue); 
+				if (matcher.find()) {
+					final String envVariableName = matcher.group(1) == null ? matcher.group(2) : matcher.group(1);
+					final String envVariableValue = System.getenv(envVariableName);
+					if (!Strings.isNullOrEmpty(envVariableValue)) {
```

Comment:
Use `System.getenv().containsKey()` to check whether an `ENV_VAR` is specified with the given name and override even if the value is an empty `String`.